src/Common/ZooKeeper/ZooKeeperImpl.cpp

```diff
@@ -562,7 +597,9 @@ void ZooKeeper::sendAuth(const String & scheme, const String & data)
     request.scheme = scheme;
     request.data = data;
     request.xid = AUTH_XID;
-    request.write(*out);
+    request.write(*maybe_compressed_out);
+    maybe_compressed_out->next();
+    out->next();
```

Comment:
how was it flushed before?

Reply:
In this place, looks like we didnt flush before. 
Also as mentioned in below comment, I will just keep one '->next'